src/resource.ts

```diff
@@ -46,42 +81,49 @@ export class Resource<ValueType> {
 
 export class ResourceWithSignal<ValueType, SourceType> {
   private fetcher: FetcherWithSource<SourceType, ValueType>;
-  private source: ReactiveValue<SourceType>;
-  private tag: Tag;
-  loading = createSignal(false);
-  error: Signal<any> = createSignal();
+  private tag = createTag();
+  private state = createSignal<State<ValueType>>(Loading());
+
+  get loading(): boolean {
+    return this.state.value[0] === LOADING_TAG;
+  }
+
+  get error(): unknown {
+    // The check is important so we never return the value for a loaded case!
+    return this.state.value[0] === ERROR_TAG ? this.state.value[1] : undefined;
+  }
 
   last?: ValueType | undefined;
 
-  current?: ValueType | undefined;
+  get current(): ValueType | undefined {
+    return this.state.value[0] === LOADED_TAG ? this.state.value[1] : undefined;
+  }
 
   constructor(
     source: ReactiveValue<SourceType>,
     fetcher: FetcherWithSource<SourceType, ValueType>
   ) {
     this.fetcher = fetcher;
-    this.source = source;
```

Comment:
I noticed that this was never reused, and so switched to just using it locally instead.